consensus/core/src/core.rs

```diff
@@ -549,7 +558,11 @@ impl Core {
     }
 
     /// Runs commit rule to attempt to commit additional blocks from the DAG.
-    fn try_commit(&mut self) -> ConsensusResult<Vec<CommittedSubDag>> {
+    /// It returns the committed sub dags and any newly accepted blocks. It is possible to have newly accepted blocks as
+    /// commits can trigger an advancement in gc round. Suspended blocks that have dependency in their causal history to any gc'ed blocks, will get unsuspended
+    /// and accepted.
+    #[instrument(level = "debug", skip_all)]
+    fn try_commit(&mut self) -> ConsensusResult<(Vec<CommittedSubDag>, Vec<VerifiedBlock>)> {
```

Comment:
The alternative I thought was `try_commit` calling directly the `add_accepted_blocks` it self for stronger guarantees if we refactor the code etc. Chose to return the accepted blocks and handle independently as I didn't want the `try_commit` do too much on its own